easybuild/tools/module_naming_scheme/toolchain.py

```diff
@@ -67,8 +67,9 @@ def det_toolchain_element_details(tc, elem):
         _log.debug("Full list of toolchain specifications: %s" % tc_ec)
     tc_ec = tc_ec[0]['ec']
     tc_deps = tc_ec['dependencies']
+    tc_hiddendeps = tc_ec['hiddendependencies']
     tc_elem_details = None
-    for tc_dep in tc_deps:
+    for tc_dep in tc_deps + tc_hiddendeps:
```

Comment:
@akesandgren: We should use `tc_ec.dependencies()` instead, which takes into account `dependencies` + `builddependencies` + `hiddendependencies` (see the `dependencies` method in `easybuild/framework/easyconfig/easyconfig.py`), and also `--filter-deps`.

And since `tc_deps` is only used in one place, there's little point to creating a variable for it:

```python
for tc_dep in tc_ec.dependencies():
    ...
```

Reply:
Well as long as there is no risk in having builddeps in there too...
But builddeps should never be part of a toolchain anyway.

Reply:
`tc_ec.dependencies` *will* include `builddependencies`... But as you say, that's irrelevant for checking toolchain definitions imho.